### PR TITLE
Recording name override

### PR DESCRIPTION
**Pull request by luciipak**

### CODE REVIEW

File: frontend/pulse3d/src/pages/upload-form.js [hunk 1]
```diff
@@ -129,17 +132,16 @@ export default function UploadForm() {
...
   const [usageModalState, setUsageModalState] = useState(false);
   const [usageModalLabels, setUsageModalLabels] = useState(modalObj.uploadsReachedDuringSession);
   const [analysisParams, setAnalysisParams] = useState(getDefaultAnalysisParams());
   const [badZipFiles, setBadZipFiles] = useState([]);
   const [resetDragDrop, setResetDragDrop] = useState(false);
   const [wellGroupErr, setWellGroupErr] = useState(false);
-  const { usageQuota } = useContext(AuthContext);
   const [creditUsageAlert, setCreditUsageAlert] = useState(false);
   const [alertShowed, setAlertShowed] = useState(false);
+  const [reanalysis, setReanalysis] = useState(false);
```
Comment (tannermpeterson) [inline frontend/pulse3d/src/pages/upload-form.js:144]: 👍

File: jobs/pulse3d/pulse3d-worker/src/main.py [hunk 2]
```diff
@@ -72,25 +74,31 @@ async def process(con, item):
             upload_details = await con.fetchrow(query, upload_id)
 
             prefix = upload_details["prefix"]
-            filename = upload_details["filename"]
-            key = f"{prefix}/{filename}"
+            metadata = json.loads(item["meta"])
+            upload_filename = upload_details["filename"]
+            name_override = metadata.get("name_override")
+            # if a new name has been given in the upload form, then replace here, else use original name
+            analysis_filename = (
```
Comment (tannermpeterson) [inline jobs/pulse3d/pulse3d-worker/src/main.py:81]: Could probably just do something like
`analysis_filename = metadata.get("name_override", upload_filename) + ".zip"`

File: jobs/pulse3d/pulse3d-worker/src/main.py [hunk 3]
```diff
@@ -133,25 +139,27 @@ async def process(con, item):
...
+                            os.path.join(tmpdir, analysis_filename),
+                            recording_df=recording_df,
+                            well_groups=well_groups,
                         )
                         recordings = list(recording)
                     except:
                         # If a user attempts to perform re-analysis on an analysis from < 0.25.2, it will fail
                         # because the parquet file won't have the raw data columns, so need to re-analyze
-                        logger.info(
+                        logger.exception(
```
Comment (tannermpeterson) [inline jobs/pulse3d/pulse3d-worker/src/main.py:162]: Can we also log the exception traceback here?

Comment (tannermpeterson) [reply jobs/pulse3d/pulse3d-worker/src/main.py:162]: Or at least just the error itself

File: jobs/pulse3d/pulse3d-worker/src/main.py (hunk 2 above)
Comment (luciipak) [reply jobs/pulse3d/pulse3d-worker/src/main.py:81]: upload_filename already includes the file extension

File: jobs/pulse3d/pulse3d-worker/src/main.py (hunk 3 above)
Comment (luciipak) [reply jobs/pulse3d/pulse3d-worker/src/main.py:162]: logger.exception auto includes the traceback

Comment (tannermpeterson) [reply jobs/pulse3d/pulse3d-worker/src/main.py:162]: Wow I did not realize that

File: jobs/pulse3d/pulse3d-worker/src/main.py (hunk 2 above)
Comment (tannermpeterson) [reply jobs/pulse3d/pulse3d-worker/src/main.py:81]: Oh true, could do `analysis_filename = f"{name_override}.zip" if (name_override := metadata.get("name_override")) else upload_filename`

**Review by tannermpeterson — APPROVED** [review]

File: jobs/pulse3d/pulse3d-worker/src/main.py (hunk 2 above)
Comment (luciipak) [reply jobs/pulse3d/pulse3d-worker/src/main.py:80]: @tannermpeterson can you approve again, I updated this var

**Review by tannermpeterson — APPROVED** [review]

**Pull request merged**